esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -1241,194 +1241,200 @@ void setDeleteFileResults(const char* fileName, const char* nodeGroup, bool fail
-        LOG(MCauditInfo, "%s,%s", auditStr, fn);
-        setDeleteFileResults(lfn, group, false, isSuper ? "Deleted Superfile" : "Deleted File", NULL, returnStr, actionResults);
-    }
-    catch(IException* e)
-    {
-        StringBuffer emsg;
-        e->errorMessage(emsg);
-        if (removeFromSuperfiles && strstr(emsg, "owned by"))
-        {
-            if (!doRemoveFileFromSuperfiles(lfn, userdesc, superFiles, failedFiles, deleteRecursively, auditStr, returnStr, actionResults))
-                return DeleteActionFailure;
-            return doDeleteFile(fn, userdesc, superFiles, failedFiles, auditStr, returnStr, actionResults, superFilesOnly, false, false);
-        }
-        if (e->errorCode() == DFSERR_CreateAccessDenied)
-            emsg.replaceString("Create ", "Delete ");
 
-        setDeleteFileResults(lfn, group, true, "Could not delete", emsg.str(), returnStr, actionResults);
-        e->Release();
-        return DeleteActionFailure;
-    }
-    catch(...)
-    {
-        setDeleteFileResults(lfn, group, true, "Could not delete", "unknown exception", returnStr, actionResults);
-        return DeleteActionFailure;
-    }
-    return DeleteActionSuccess;
-}
+            auditStr.set(",FileAccess,WsDfu,DELETED,");
+            auditStr.append(espProcess.get());
+            auditStr.append(',');
+            if (!isEmptyString(user))
+                auditStr.append(user).append('@');
+            context.getPeer(auditStr);
+        }
 
-void doDeleteFiles(StringArray &files, IUserDescriptor *userdesc, StringArray &superFiles, StringArray &failedFiles,
-    const char *auditStr, StringBuffer &returnStr, IArrayOf<IEspDFUActionInfo> &actionResults,
-    bool superFilesOnly, bool removeFromSuperfiles, bool deleteRecursively)
-{
-    ForEachItemIn(i, files)
-    {
-        const char* fn = files.item(i);
-        if(!fn || !*fn)
-            continue;
+        const char *getReturnStr() const { return returnStr.str(); }
+        IArrayOf<IEspDFUActionInfo> &getActionResults() { return actionResults; }
 
-        PROGLOG("Deleting %s", fn);
-        if (DeleteActionFailure==doDeleteFile(fn, userdesc, superFiles, failedFiles, auditStr, returnStr, actionResults, superFilesOnly, removeFromSuperfiles, deleteRecursively))
+        void deleteFiles(StringArray &logicalFileNames, bool removeFromSuperFiles, bool removeEmptyOwningSuperFiles)
         {
-            failedFiles.appendUniq(fn);
-            PROGLOG("Delete %s failed", fn);
-        }
-        else
-            PROGLOG("Delete %s done", fn);
-    }
+            StringArray emptyOwningSuperFiles;
+            IDistributedFileDirectory &fdir = queryDistributedFileDirectory();
+            ForEachItemIn(i, logicalFileNames)
+            {
+                const char *fn = logicalFileNames.item(i);
+                if (emptyOwningSuperFiles.contains(fn))
+                    continue;
 
-}
+                CDfsLogicalFileName dfsLFN;
+                dfsLFN.set(fn);
```

Comment:
should move this up and set 'fn' to dfsLfn.get() , so that the name is normalized.
Otherwise, the existing code could miss matches when doing emptyOwningSuperFiles.contains(fn).

Reply:
If the original logical file name has group information, will dfsLfn.get() still keep that information? Which name (original logical file name or dfsLfn.get()) should be used in line 1414: deleteFile(fn)?

Reply:
I am going to move this up and use the lfn to emptyOwningSuperFiles.contains(). I may keep the line 1414 as is.